core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -599,6 +599,9 @@ public final NodeChannels openConnection(DiscoveryNode node, ConnectionProfile c
                 nodeChannels = new NodeChannels(nodeChannels, version); // clone the channels - we now have the correct version
                 transportService.onConnectionOpened(nodeChannels);
                 connectionRef.set(nodeChannels);
+                if (!Arrays.stream(nodeChannels.channels).allMatch(this::isOpen)) {
```

Comment:
can we use `== false`?

Reply:
I pushed 1445272de528681cd7505752bb19d1f771b2193d.